### PR TITLE
[WIP] fix: decimal step cause divide not equal

### LINKED ISSUE #227: Pushable only works in positive direction with decimal steps
Not sure if this is a proper bug really, since maybe it shouldn't be possible to use decimals? I'm reporting it in any case:

The following example should demonstrate an issue where if you drag the leftmost handle to the right you can push the other handles, but dragging the rightmost handle to the left does not let you push the others.

http://codepen.io/oddsor/pen/vxOLVO

I'm not sure if this bug is known or if it's something that is easily fixable, so I've just worked around it by ensuring that I use "integers".

### PR DESCRIPTION
**Pull request by paranoidjk**
close #227

### CODE REVIEW

**Comment by paranoidjk** [comment]
ping @benjycui

File: src/Range.jsx [hunk 1]
```diff
@@ -176,10 +176,17 @@ class Range extends React.Component {
     const { marks, step, min, max } = this.props;
     const cache = this._getPointsCache;
     if (!cache || cache.marks !== marks || cache.step !== step) {
+      const decimal = step.toString().split('.')[1] && step.toString().split('.')[1].length;
```
Comment (benjycui) [inline src/Range.jsx]: Use [`getPrecision`](https://github.com/react-component/slider/blob/4e26da531406305e1c2b9396d13c84c024193fba/src/utils.js#L28-L35) ？

File: src/Range.jsx [hunk 2]
```diff
@@ -176,10 +176,17 @@ class Range extends React.Component {
...
       const pointsObject = { ...marks };
       if (step !== null) {
-        for (let point = min; point <= max; point += step) {
+        let point = min;
+        while (point <= max) {
```
Comment (benjycui) [inline src/Range.jsx]: This loop will cost a lot.

e.g. min = 0 && max = 99999

File: src/Range.jsx (hunk 1 above)
Comment (paranoidjk) [inline src/Range.jsx]: ok

File: src/Range.jsx (hunk 2 above)
Comment (benjycui) [inline src/Range.jsx]: https://github.com/react-component/slider/issues/46

Comment (paranoidjk) [inline src/Range.jsx]: the performance problem only occur when users set `ste` to be a very small decimals, I think it's not our worries ?

Comment (paranoidjk) [inline src/Range.jsx]: maybe use something like `binary search` + `Array.map` to caculate the point more quickly?

Comment (paranoidjk) [inline src/Range.jsx]: ~~caculate the point more quickly~~ ——> avoid Maximum call stack

Comment (paranoidjk) [inline src/Range.jsx]: Another idea,  what if when `step` is very small compare to `max - min`, we don't calculate the whole points array at first, instead just get the formula，something like： `value = f(index)`, then the cost just happen when every time getter

**Comment by paranoidjk** [comment]
close. going to open a new pr to fix the decimal step problem.

**Pull request closed without merging**